### PR TITLE
nezha-agent: init at 0.16.1

### PR DESCRIPTION
**Pull request by Moraxyc**
Add nezha-agent to pkgs

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by Moraxyc** [comment]
Result of `nixpkgs-review pr 281278` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
  </ul>
</details>
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>nezha-agent</li>
  </ul>
</details>

File: pkgs/by-name/ne/nezha-agent/package.nix [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  vendorHash = "sha256-b8jICFsQaSetBzhTiq8zBX/LYuhQNly5kj45ZV6NDiY=";
+
+  ldflags = [
+    "-s"
+    "-w"
+    "-X main.version=${version}"
+  ];
+
+  preCheck = ''
+    rm ./pkg/monitor/myip_test.go
```
Comment (SuperSandro2000) [inline pkgs/by-name/ne/nezha-agent/package.nix:28]: I assume the test fails but a comment would be good to document this.

File: nixos/modules/services/monitoring/nezha-agent.nix [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+{
+  meta = {
+    maintainers = with lib.maintainers; [ moraxyc ];
+  };
+  options = {
+    services.nezha-agent = {
+      enable = lib.mkEnableOption (lib.mdDoc "Agent of Nezha Monitoring");
+
+      package = lib.mkPackageOption pkgs "nezha-agent" { };
+      debug = lib.mkOption {
```
Comment (SuperSandro2000) [inline nixos/modules/services/monitoring/nezha-agent.nix:14]: We could use mkEnableOption here, too.

File: nixos/modules/services/monitoring/nezha-agent.nix [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+        '';
+      };
+      passwordFile = lib.mkOption {
+        type = with lib.types; nullOr str;
+        default = null;
+        description = lib.mdDoc ''
+          Path to the file contained the password from dashboard.
+        '';
+      };
+      password = lib.mkOption {
```
Comment (SuperSandro2000) [inline nixos/modules/services/monitoring/nezha-agent.nix:64]: I don't think we should offer an insecure option to set the passowrd.

File: pkgs/by-name/ne/nezha-agent/package.nix (hunk 1 above)
Comment (Moraxyc) [reply pkgs/by-name/ne/nezha-agent/package.nix:29]: This test makes a request to get geoip data to test whether it works correctly. But we build it in the sandbox without network.

File: nixos/modules/services/monitoring/nezha-agent.nix (hunk 3 above)
Comment (Moraxyc) [reply nixos/modules/services/monitoring/nezha-agent.nix]: Removed now

**Review by SuperSandro2000 — APPROVED** [review]

**Pull request merged**